### PR TITLE
Remove stray DNS port references in NetPol tests

### PR DESCRIPTION
**Pull request by danwinship**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
The NetPol tests no longer use DNS (#102354), so we shouldn't be adding DNS-related exceptions to the policies that are being tested.

#### Which issue(s) this PR fixes:
none

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/sig network
/priority backlog
cc @jayunit100

### CODE REVIEW

**Comment by jayunit100** [comment]
/retest

63 wrong results in one of the matrices, retrying if fails again will check

File: test/e2e/network/netpol/network_policy.go [hunk 1]
```diff
@@ -686,10 +683,6 @@ var _ = common.SIGDescribe("Netpol", func() {
 							// don't use named ports
 							Port: &intstr.IntOrString{Type: intstr.Int, IntVal: 80},
 						},
-						{
```
Comment (jayunit100) [inline test/e2e/network/netpol/network_policy.go:689]: note to self in case we ever revitalize dns policy tests this needs to be added back :)

**Comment by jayunit100** [comment]
not against this, but curious is this actually something that causes a problem? or just cleanup ?

**Comment by knabben** [comment]
@danwinship does it makes sense to keep at least one test case-specific for DNS? (outside this PR)

**Comment by knabben** [comment]
/retest

last fail is not related with the PR

**Comment by jayunit100** [comment]
```
Kubernetes e2e suite: [sig-network] Netpol [LinuxOnly] NetworkPolicy between server and client using UDP should support a 'default-deny-ingress' policy [Feature:NetworkPolicy] expand_more
--
```

**Comment by knabben** [comment]
can be very tricky to debug, wondering if it's possible to fetch calico logs in the artifacts and correlate the netpol creation time.

**Comment by jayunit100** [comment]
https://github.com/kubernetes/kubernetes/pull/104086 wlil reduce cpu usage by 2/3 or more, that should squash any flakes at high concurrency  will test today

**Comment by danwinship** [comment]
> not against this, but curious is this actually something that causes a problem? or just cleanup ?

Just cleanup. I was just trying to confirm that OpenShift's "rebase to 1.22" PR included #102354, and got confused by the fact that there were still lots of references to port 53 in the code.

> note to self in case we ever revitalize dns policy tests this needs to be added back :)

well, no, I guess that's the other reason for removing it; because if you try to add it back in the same way it was before, you're just going to bring back #102286.

> does it makes sense to keep at least one test case-specific for DNS?

DNS is not special; we don't need to test both "a policy that allows traffic to a specific UDP port" _and_ "a policy that allows traffic to DNS".

I guess it's "special" in that we'd like for it to be possible to say "allow egress to DNS" in a generic way, but you can't actually do that. But that's a design problem not a testing problem.

**Pull request merged**